src/lamp_py/ingestion/convert_gtfs_rt.py

```diff
@@ -353,38 +343,151 @@ def gz_to_pyarrow(
             table,
         )
 
-    def write_table(self, table: pyarrow.table) -> None:
-        """write the table to our s3 bucket"""
-        # don't write if table has no data
-        if table.num_rows == 0:
-            return
+    def verify_table(self, table: pyarrow.Table) -> None:
+        """
+        verify structure of table
+        """
+        partition_cols = ["year", "month", "day", "hour"]
+        for col in partition_cols:
+            unique_list = pc.unique(table.column(col)).to_pylist()
 
-        try:
-            s3_prefix = str(self.config_type)
+            assert (
+                len(unique_list) == 1
+            ), f"{self.config_type} Table column {col} had {len(unique_list)} unique elements"
 
-            sort_log = ProcessLogger(
-                "pyarrow_sort_by", table_rows=table.num_rows
+    def sync_with_s3(self, part_folder: str) -> str:
+        """
+        check and see if UUID path exists on s3
+        """
+
+        local_folder = os.path.join(
+            self.tmp_folder,
+            part_folder,
+        )
+        if os.path.exists(local_folder):
+            return os.path.join(local_folder, os.listdir(local_folder)[0])
+
+        os.makedirs(local_folder, exist_ok=True)
+        bucket = os.environ["SPRINGBOARD_BUCKET"]
+        s3_files = file_list_from_s3(
+            bucket,
+            file_prefix=f"{part_folder}/",
+        )
+        if len(s3_files) == 1:
+            s3_path = s3_files[0].replace("s3://", "")
+            local_path = s3_path.replace(bucket, self.tmp_folder)
+            download_file(s3_path, local_path)
+            return local_path
+
+        return ""
+
+    # pylint: disable=R0914
+    def continuous_pq_update(self, table: pyarrow.Table) -> None:
+        """
+        Continuous updating of local parquet files that are synced with S3
+        """
+        log = ProcessLogger("continuous_pq_update")
+        log.log_start()
+        try:
+            self.verify_table(table)
+
+            year = pc.unique(table.column("year")).to_pylist()[0]
+            month = pc.unique(table.column("month")).to_pylist()[0]
+            day = pc.unique(table.column("day")).to_pylist()[0]
+            hour = pc.unique(table.column("hour")).to_pylist()[0]
```

Comment:
i could see these being put into a single function, `extract_datetime` that asserts if they are not all the same and returns a tuple of datetime ints.

Reply:
this has been partially incorporated, a datetime object (representing partition date info) is now returned from the function that verifies unique table values